### PR TITLE
feat[asana]: ability to emit events only for some fields

### PR DESCRIPTION
**Pull request by gsomoza**
NOTES: only standard fields are supported right now (no custom fields). Standard fields are retrieved from the first task available in the project.

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/asana/asana.app.mjs [hunk 1]
```diff
@@ -117,6 +117,29 @@ export default {
...
+
+        if (!tasks || tasks.length === 0) {
+          return [];
+        }
+        const task = await this.getTask(tasks[0].gid);
+
+        return Object.keys(task).map((key) => ({
+          label: key,
+          value: key,
+        }));
```
Comment (GTFalcao) [inline components/asana/asana.app.mjs:140]: If label and value are the same, you can use a simple array of strings instead:
```suggestion
        return Object.keys(task);
```

File: components/asana/sources/task-field-updated-in-project/task-field-updated-in-project.mjs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        (c) => ({
+          project: c.project,
+        }),
+      ],
+    },
+  },
+  methods: {
+    ...common.methods,
+    getWebhookFilter() {
+      console.log("Using filters: " + this.taskFields);
```
Comment (GTFalcao) [inline components/asana/sources/task-field-updated-in-project/task-field-updated-in-project.mjs:39]:
```suggestion
```

File: components/asana/sources/task-field-updated-in-project/task-field-updated-in-project.mjs [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import common from "../common/common.mjs";
+const { asana } = common.props;
+
+export default {
+  ...common,
+  key: "asana-task-field-updated-in-project",
+  type: "source",
+  name: "Task Field Updated In Project (Instant)",
+  description: "Emit a new event whenever a given task field is updated.",
+  version: "1.0.0",
```
Comment (GTFalcao) [inline components/asana/sources/task-field-updated-in-project/task-field-updated-in-project.mjs:10]:
```suggestion
  version: "0.0.1",
```

**Review by GTFalcao — COMMENTED** [review]
Hi @gsomoza , nice work on yet another improvement!

I left a couple of simple comments. Once again it is needed to update the package.json version, and it'll probably be easier to keep track of that if you combine all your improvements into a single branch/pull request.

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

**Pull request closed without merging**